pkg/menus/menus.go

```diff
@@ -0,0 +1,174 @@
+package menus
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+var menuWidth = 50
+
+// Entry contains all the information needed for a boot entry
+type Entry interface {
+	// Label is the string will show in menu
+	Label() string
+	// IsDefault is true means this entry will be hit by default.
+	// If there is many default choise, the first on in the list will be choose.
+	// the rest will be ignored.
+	IsDefault() bool
+	Do() error
+}
+
+// DisplayMenu will present all entries into a menu with numbers, so that user can choose from them.
+func DisplayMenu(menuTitle string, introwords string, location int, entries []Entry) (Entry, error) {
```

Comment:
Note: for future, it'd be nice to be able to go 'back' in this menu flow, maybe we can open an issue for that and get to it later